bundles/CoreBundle/Command/ThumbnailsImageCommand.php

```diff
@@ -113,30 +113,42 @@ protected function fetchItems(InputInterface $input): array
 
         $list->setCondition(implode(' AND ', $conditions));
 
-        return $list->loadIdList();
+        $assetIdsList = $list->loadIdList();
+        $thumbnailList = new Asset\Image\Thumbnail\Config\Listing();
+
+        $allowedThumbs = [];
+        if ($input->getOption('thumbnails')) {
+            $allowedThumbs = explode(',', $input->getOption('thumbnails'));
+        }
+
+        $items = [];
+        foreach ($assetIdsList as $assetId) {
+            foreach ($thumbnailList->getThumbnails() as $thumbnailConfig) {
+                $thumbName = $thumbnailConfig->getName();
+                if (empty($allowedThumbs) || in_array($thumbName, $allowedThumbs)) {
+                    $items[] = $assetId . '~~~' . $thumbName;
+                }
+            }
+        }
+
+        return $items;
     }
 
-    protected function runSingleCommand(string $assetId, InputInterface $input, OutputInterface $output): void
+    protected function runSingleCommand(string $item, InputInterface $input, OutputInterface $output): void
     {
+        list($assetId, $thumbnailConfigName) = explode('~~~', $item);
```

Comment:
Maybe add a third parameter to ensure the result will be a tuple:
```suggestion
        list($assetId, $thumbnailConfigName) = explode('~~~', $item, 2);
```

Reply:
makes sense! thanks 👍